orc8r/cloud/go/services/certifier/servicers/certifier.go

```diff
@@ -768,6 +814,48 @@ func checkAction(reqResource *certprotos.Resource, policyResource *certprotos.Re
 	return certprotos.Effect_UNKNOWN
 }
 
+// checkResource checks if the requested resource is authorized by the policy
+func checkResource(reqResource *certprotos.RequestResource, policyResource *certprotos.PolicyResource) certprotos.Effect {
+	// The policy does not handle this case, so return UNKNOWN.
+	if checkResourceType(reqResource, policyResource) {
+		return certprotos.Effect_UNKNOWN
+	}
+
+	finalEffect := certprotos.Effect_UNKNOWN
+	if path := policyResource.GetPath(); path != nil {
+		if ok, _ := doublestar.Match(path.Path, reqResource.Resource); ok {
+			finalEffect = policyResource.Effect
+		}
+	}
+
+	reqResourceType := getRequestResourceType(reqResource)
+	policyResourceType := getPolicyResourceType(policyResource)
+	matched := false
+	if reqResourceType == constants.NetworkID && policyResourceType == constants.NetworkID {
+		reqID := reqResource.GetNetworkId()
+		policyIDs := policyResource.GetNetwork().Networks
+		for _, policyID := range policyIDs {
+			if reqID == policyID {
+				matched = true
+			}
+		}
+	}
+	if reqResourceType == constants.TenantID && policyResourceType == constants.TenantID {
+		reqID := reqResource.GetTenantId()
+		policyIDs := policyResource.GetTenant().Tenants
+		for _, policyID := range policyIDs {
+			if reqID == policyID {
+				matched = true
+			}
+		}
+	}
+
+	if finalEffect != certprotos.Effect_DENY && matched {
+		finalEffect = policyResource.Effect
+	}
+	return finalEffect
```

Comment:
Why the "matched" var etc? Is a little confusing / unclear why not just do an early return like for the path-based resources

Reply:
Yup I agree, removing it 👍